CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
```

Comment:
Any reason why the functions in this class cannot be member functions instead of we making them static?

Reply:
It was done for the sake of consistency, as the other listerner class `CRM_Civicase_Event_Listener_ActivityFilter` also has static methods only.
Maybe it's okay to member methods, but are you sure it would work?
```
Civi::dispatcher()->addListener('civi.api.respond', ['CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler', 'onRespond'], 10);
```

Reply:
Static functions should be avoided as much as possible except when necessary. It defeats the purpose of OOP. You need to test to see if it works, I see no reason why it shouldn't except Civi has made it mandatory to make listener classes have static methods

Reply:
Okay will test.